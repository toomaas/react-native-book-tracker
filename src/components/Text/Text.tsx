import {useTheme} from '@react-navigation/native';
import React from 'react';
import {Text as RNText} from 'react-native';
import {TextProps} from './Text.types';

const Text: React.FunctionComponent<TextProps> = props => {
  const {style} = props;

  const {colors} = useTheme();

  return (
    <RNText {...props} style={[style, {color: colors.text}]}>
      {props.children}
    </RNText>
  );
};

export default Text;
