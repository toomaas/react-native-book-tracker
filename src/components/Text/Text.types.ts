import {TextProps as RNTextProps} from 'react-native';

export type TextProps = RNTextProps;
