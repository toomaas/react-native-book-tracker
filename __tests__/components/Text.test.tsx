import renderer from 'react-test-renderer';
import Text from '../../src/components/Text';
import React from 'react';

describe('Text Component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
