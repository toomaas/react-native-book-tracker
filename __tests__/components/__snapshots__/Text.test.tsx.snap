// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "color": "rgb(28, 28, 30)",
      },
    ]
  }
/>
`;
